includes/compat/class-customer.php

```diff
@@ -0,0 +1,170 @@
+<?php
+/**
+ * Backwards Compatibility Handler for Customers.
+ *
+ * @package     EDD
+ * @subpackage  Compat
+ * @copyright   Copyright (c) 2018, Easy Digital Downloads, LLC
+ * @license     http://opensource.org/licenses/gpl-2.0.php GNU Public License
+ * @since       3.0.0
+ */
+namespace EDD\Compat;
+
+// Exit if accessed directly
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * Customer Class.
+ *
+ * @since 3.0.0
+ */
+class Customer extends Base {
+
+	/**
+	 * Constructor.
+	 *
+	 * @since 3.0.0
+	 */
+	public function __construct() {
+		$this->set_component( 'customer' );
```

Comment:
Why not just adjust `Base` to inject a required `$component` via a constructor then set `$component there directly? Making subclasses set things that _every_ subclass needs to set just encourages unnecessary duplication, imo.